app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,17 @@
+class CommentsController < ApplicationController
+  def create
+    @p_author = User.find(params[:user_id])
+    @post = Post.find(params[:post_id])
+    @comment = Comment.new(params.require(:comment).permit(:text))
+    @comment.author = current_user
+    @comment.post = @post
+
+    if @comment.save
+      flash[:success] = 'Comment saved successfully.'
+      redirect_to user_post_path(@p_author, @post)
+    else
+      flash.now[:error] = 'Error: Comment could not be saved.'
+      render :new
+    end
+  end
+end
```

Comment:
- [ ] The comments controller should have new and create actions. Only the create is existing.

```ruby
def new
    @comment = Comment.new
end
```

- [OPTIONAL] You can make use of the strong parameters that I suggested in my previous comment.

```ruby
private 
def comment_params
   params.require(:comment).permit(:text)
end
```